### PR TITLE
Add tests to ensure certain pods appear on infra nodes

### PR DESCRIPTION
**Pull request by NautiluX**
* test if infra nodes exist (exactly 2)
* test if following pods are scheduled to infra nodes:
  * sre-machine-api-status-exporter
  * sre-ebs-iops-reporter
  * sre-stuck-ebs-vols

### CODE REVIEW

**Comment by NautiluX** [comment]
/hold
we need to wait for the changes to the operators are promoted before merging

File: test/verify/infranodes.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	)
+
+	BeforeEach(func() {
+		listNodesOptions := metav1.ListOptions{}
+		listNodesOptions.LabelSelector = "node-role.kubernetes.io/infra="
+		nodeList, nodeListErr = h.Kube().CoreV1().Nodes().List(listNodesOptions)
+		infraNodeAddresses = getInfraNodeAddresses(nodeList)
+	})
+
+	It("exactly 2 exist", func() {
```
Comment (jeefy) [inline test/verify/infranodes.go]: Is this the case for every OSD cluster?

Comment (NautiluX) [reply test/verify/infranodes.go]: AFAIK this is hardcoded currently

Comment (NautiluX) [reply test/verify/infranodes.go]: it is. Confirmed with @oourfali

Comment (jeefy) [reply test/verify/infranodes.go]: kk thanks

**Comment by meowfaceman** [comment]
/approve
This looks good to me, but just want to confirm that the other related PRs have been pushed before merging.

**Comment by NautiluX** [comment]
yes they are merged.

**Pull request merged**